buildSrc/src/main/kotlin/otel.japicmp-conventions.gradle.kts

```diff
@@ -44,6 +45,21 @@ class AllowDefaultMethodRule : AbstractRecordingSeenMembers() {
         // change.
         continue
       }
+      if (!change.isSourceCompatible) {
+        if (member is JApiMethod &&
+          member.getjApiClass().getFullyQualifiedName() == "io.opentelemetry.sdk.testing.trace.TestSpanData" &&
+          member.getName() == "getInstrumentationScopeInfo"
```

Comment:
The need for this is actually a weird situation that I think requires some discussion.

The method triggering this is compatibility violation is [TestSpanData#getInstrumentationScopeInfo](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/testing/src/main/java/io/opentelemetry/sdk/testing/trace/TestSpanData.java#L67). Its has a [default](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java#L169-L171) implementation on SpanData, which I overrode as abstract again in `TestSpanData` so that autovalue would accept it as a constructor argument. 

jApiCmp doesn't like this because its counts as a `METHOD_ABSTRACT_ADDED_TO_CLASS`. However, context matters, and in this case its not actually a breaking change because the package private constructor of `TestSpanData` prevents implementations of `TestSpanData` besides from autovalue. 

We actually a bit stuck as we are very likely to see this pattern again given our use of autovalue, and each time it will be a jApiCmp violation.

Reply:
is there a way to skip this check for classes annotated with autovalue?

Reply:
In general, if we want to be able to conform to japcmp's rule that adding methods to abstract classes (i.e. `METHOD_ABSTRACT_ADDED_TO_CLASS`) with package private constructors counts as a breaking change, then we shouldn't use `@Autovalue` for classes part of our public API.

Reply:
I couldn't find any obvious way to access the classes annotations from this code.

Reply:
#4650 shows how transitioning `TestSpanData` to a concrete class can solve this problem. We could transition public `@Autovalue` classes to concrete instances as needed going forward.

Reply:
right, new AutoValues will have this problem. AFAIK it's possible to access annotations here, let me check

Reply:
I was able to filter out incompatibilities for abstract methods on any AutoValue classes.